vagrant/build.sh

```diff
@@ -62,24 +70,17 @@ else
 fi
 
 # Create maestro dummy config if config does not exist
-[ -f maestro.config ] || \
-echo "network:
-    interfaces:
-        - if_name: eth1
-          exists: replace
-          dhcpv4: true
-          hw_addr: \"{{ARCH_ETHERNET_MAC}}\"
-        - if_name: eth2
-          exists: replace
-          dhcpv4: false
-          ipv4_addr: 10.0.102.102
-          ipv4_mask: 24
-          hw_addr: \"{{ARCH_ETHERNET_MAC}}\"
-devicedb_conn_config:
-    devicedb_uri: \"https://$DEVICE_ID:9090\"
-    devicedb_prefix: \"vagrant\"
-    devicedb_bucket: \"lww\"
-    relay_id: \"$DEVICE_ID\"
-    ca_chain: \"$DEVICEDB_SRC/hack/certs/myCA.pem\"
-config_end: true
-" >> maestro.config
+if [ ! -f maestro.config ]; then
+    SYMPHONY_HOST="gateways.mbedcloudstaging.net"
+    SYMPHONY_CLIENT_CRT=$(cat $MAESTRO_CERTS/device_cert.pem)
+    SYMPHONY_CLIENT_KEY=$(cat $MAESTRO_CERTS/device_private_key.pem)
+
+    mv /tmp/maestro.config maestro.config
+    sed -i "s|{{DEVICE_ID}}|$DEVICE_ID|g" maestro.config
+    sed -i "s|{{DEVICEDB_SRC}}|$DEVICEDB_SRC|g" maestro.config
+    sed -i "s|{{SYMPHONY_HOST}}|$SYMPHONY_HOST|g" maestro.config
```

Comment:
why do we employ 2 different methods of modifying the maestro config?  Could we use yq for this section too?

Reply:
`DEVICE_ID` needs to be replaced in more than 1 location - so I think that should be a `sed` command.
`DEVICEDB_SRC` may be replaced in more than 1 location in the future, so same as above. Currently only 1 replacement though.

`SYMPHONY_HOST` definitely can be a `yq` command though. I will fix that

Reply:
I'll ask the counter question.

Was there any specific reason why `yq` was needed that GNU Tools couldn't have solved?
Imo, adding another apt repository for a single tool feels expensive.

Reply:
Which GNU tool can intelligently parse yaml?  As shown here, sed can work in some limited cases, but it had problems with multi-line values and resulted in a broken yaml file.

Reply:
While all of this is achievable with sed/awk/tr, I think cost is the lower benefit here versus readability. Vagrant is a development/test environment, so there are no embedded constraints per se (disk/memory/setup). On the other hand, having to manually take care of indentation with sed is prone to breakage and introduces additional maintenance, e.g. if the yaml file is changed independently and one of the above values suddenly requires a different indentation level.

At the same time, using sed for some of the replacements also makes sense because it is the more suited tool to use (e.g. for multiple replacements and partial value/substring replacements).